src/d/actor/d_a_steam_tag.cpp

```diff
@@ -4,56 +4,209 @@
 }
 
 /* 000003BC-000003DC       .text daSteamTag_Execute__FP12daSteamTag_c */
-static BOOL daSteamTag_Execute(daSteamTag_c*) {
-    /* Nonmatching */
+static BOOL daSteamTag_Execute(daSteamTag_c* i_this) {
+    return ((daSteamTag_c*)i_this)->execute();
 }
 
 /* 000003DC-000006FC       .text execute__12daSteamTag_cFv */
 BOOL daSteamTag_c::execute() {
-    /* Nonmatching */
+    float windStrength;
+    float fVar7;
+    
+    if ((mEmitTimer == 0) && (mCreateTimer > 0)) {
+        mCreateTimer--;
+        if (mCreateTimer == 0) {
+            if (createEmitter()) {
+                JGeometry::TVec3<s16> angle;
+                angle.x = current.angle.x;
+                angle.y = current.angle.y;
+                angle.z = current.angle.z;
+                mpEmitter->setGlobalTranslation(current.pos);
+                mpEmitter->setGlobalRotation(angle);
+                mpEmitter->setGlobalAlpha(getData()->mSteamAlpha);
+                mpEmitter->playCreateParticle();
+                mEmitTimer = getData()->mEmitTimer_offset + cM_rndF(getData()->mEmitTimer_seed);
+            }
+            else {
+                mCreateTimer = 0x1e;
+            }
+        }
+    }
+    else if ((mCreateTimer == 0) && (mEmitTimer > 0)) {
+        mEmitTimer--;
+        if (mEmitTimer == 0) {
+            if (mpEmitter != NULL) {
+                windStrength = 0.0f;
+                mpEmitter->stopCreateParticle();
+                if (mpEmitter->getParticleNumber() == 0) {
+                    mpEmitter->becomeInvalidEmitter();
+                    endEmitter();
+                    mpEmitter = NULL;
+                    mCreateTimer = getData()->mCreateTimer_offset + cM_rndF(getData()->mCreateTimer_seed);
+                }
+                else {
+                    mEmitTimer = 1;
+                    windStrength = 1.0f;
+                }
+                mPointWind.set_pwind_power(windStrength);
+            }
+        }
+        else {
+            fopAcM_seStart(this, JA_SE_OBJ_STEAM, 0);
+            if (mpEmitter != NULL) {
+              mpEmitter->setGlobalAlpha(getData()->mSteamAlpha);
+            }
+            mCps.cM3dGCps::Set(mPntWindCpsS);
+            dComIfG_Ccsp()->Set(&mCps);
+            mPointWind.set_pwind_move();
+        }
+    }
+    return TRUE;
 }
 
 /* 000006FC-00000704       .text daSteamTag_IsDelete__FP12daSteamTag_c */
 static BOOL daSteamTag_IsDelete(daSteamTag_c*) {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00000704-0000072C       .text daSteamTag_Delete__FP12daSteamTag_c */
-static BOOL daSteamTag_Delete(daSteamTag_c*) {
-    /* Nonmatching */
+static BOOL daSteamTag_Delete(daSteamTag_c* i_this) {
+    ((daSteamTag_c*)i_this)->~daSteamTag_c();
+    return TRUE;
+}
+
+daSteamTag_c::~daSteamTag_c() {
+    if (mpEmitter != NULL) {
+      mpEmitter->becomeInvalidEmitter();
+      mpEmitter = NULL;
+    }
+    mPointWind.set_pwind_delete();
+    return;
+}
+
+s32 daSteamTag_c::create() {
+    int result;
```

Comment:
```suggestion
    int phase_state;
```
We know the exact name they liked to use for this local variable because it appears in debug asserts for dozens of actors